airbyte-workers/src/main/java/io/airbyte/workers/internal/EmptyAirbyteSource.java

```diff
@@ -5,29 +5,79 @@
+import io.airbyte.config.ResetSourceConfiguration;
+import io.airbyte.config.StateType;
+import io.airbyte.config.StateWrapper;
+import io.airbyte.config.StreamDescriptor;
 import io.airbyte.config.WorkerSourceConfig;
+import io.airbyte.config.helpers.StateMessageHelper;
+import io.airbyte.protocol.models.AirbyteGlobalState;
 import io.airbyte.protocol.models.AirbyteMessage;
 import io.airbyte.protocol.models.AirbyteMessage.Type;
 import io.airbyte.protocol.models.AirbyteStateMessage;
+import io.airbyte.protocol.models.AirbyteStateMessage.AirbyteStateType;
+import io.airbyte.protocol.models.AirbyteStreamState;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.EmptyStackException;
+import java.util.HashSet;
+import java.util.LinkedList;
 import java.util.Optional;
+import java.util.Queue;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * This source will never emit any messages. It can be used in cases where that is helpful (hint:
  * reset connection jobs).
  */
+@Slf4j
 public class EmptyAirbyteSource implements AirbyteSource {
 
   private final AtomicBoolean hasEmittedState;
+  private final Queue<StreamDescriptor> streamDescriptors = new LinkedList<>();
+  private boolean isResetBasedOnConfig;
+  private boolean isStarted = false;
+  private Optional<StateWrapper> stateWrapper;
 
   public EmptyAirbyteSource() {
     hasEmittedState = new AtomicBoolean();
   }
 
   @Override
   public void start(final WorkerSourceConfig sourceConfig, final Path jobRoot) throws Exception {
-    // no op.
+
+    try {
+      if (sourceConfig == null || sourceConfig.getSourceConnectionConfiguration() == null) {
+        isResetBasedOnConfig = false;
+      } else {
+        final ResetSourceConfiguration sourceConfiguration =
+            Jsons.object(sourceConfig.getSourceConnectionConfiguration(), ResetSourceConfiguration.class);
+        streamDescriptors.addAll(sourceConfiguration.getStreamsToReset());
+        if (streamDescriptors.isEmpty()) {
+          // TODO: This is done to be able to handle the transition period where we can have no stream being pass to the configuration because the
+          //  logic of populating this list is not implemented
+          isResetBasedOnConfig = false;
+        } else {
+          stateWrapper = StateMessageHelper.getTypedState(sourceConfig.getState().getState());
+
+          if (stateWrapper.isPresent() &&
+              stateWrapper.get().getStateType() == StateType.LEGACY &&
+              !configHasFullCatalog(sourceConfig)) {
+            log.error("The state a legacy one but we are trying to do a partial update, this is not supported.");
+            throw new IllegalStateException("Try to perform a partial reset on a legacy state");
+          }
+
+          isResetBasedOnConfig = true;
+        }
+      }
+    } catch (final IllegalArgumentException e) {
```

Comment:
what throws this illegal argument exception? there's 20 lines of code in the try block so it is hard to figure out what this comes from and what it means. can it go into a helper method please?